lib/compat/wordpress-6.3/script-loader.php

```diff
@@ -13,3 +13,69 @@
 remove_action( 'wp_body_open', 'wp_global_styles_render_svg_filters' );
 remove_action( 'in_admin_header', 'wp_global_styles_render_svg_filters' );
 
+/**
+ * Collect the block editor assets that need to be loaded into the editor's iframe.
+ *
+ * @since 6.0.0
+ * @access private
+ *
+ * @return array {
+ *     The block editor assets.
+ *
+ *     @type string|false $styles  String containing the HTML for styles.
+ *     @type string|false $scripts String containing the HTML for scripts.
+ * }
+ */
+function _gutenberg_get_iframed_editor_assets() {
+	global $wp_styles, $wp_scripts;
+
+	// Keep track of the styles and scripts instance to restore later.
+	$current_wp_styles  = $wp_styles;
+	$current_wp_scripts = $wp_scripts;
+
+	// Create new instances to collect the assets.
+	$wp_styles  = new WP_Styles();
+	$wp_scripts = new WP_Scripts();
+
+	// Register all currently registered styles and scripts. The actions that
+	// follow enqueue assets, but don't necessarily register them.
+	$wp_styles->registered  = $current_wp_styles->registered;
+	$wp_scripts->registered = $current_wp_scripts->registered;
+
+	wp_enqueue_style( 'wp-block-editor-content' );
```

Comment:
Enqueuing them before dispatching the `enqueue_block_assets` action does result in these stylesheets being loaded before the consumer-provided styles, which is expected. However, because the `wp-block-library` is loaded by the compat-layer ([see](https://github.com/WordPress/gutenberg/pull/49655/files#r1176697135)) this one is loaded _after_ the consumer provided styles.

We can fix this by declaring the `wp-block-library` a dependency of the `wp-block-editor-content`, so it loads before the assets enqueued via `enqueue_block_assets` and the compat-layer. Sent a suggestion at https://github.com/WordPress/gutenberg/pull/49655/files#r1176713569